drivers/net/mdio/of_mdio.c

```diff
@@ -112,7 +112,13 @@ static int of_mdiobus_register_phy(struct mii_bus *mdio,
 	bool is_c45;
 	int rc;
 	u32 phy_id;
+	char child_name[64];
```

Comment:
since there is no kernel printing when of_mdiobus_register_phy I have added start+end prints (otherwise it would be hard to check this change), 
char child_name[64];  purpose is to save the child name with the protection that child is not null.
I can replace it with this if you think it is better:
printk(KERN_INFO "start of_mdiobus_register_phy. name=%s\n", child ? child->name, "null");
printk(KERN_INFO "end of_mdiobus_register_phy. name=%s\n", child ? child->name, "null");

Reply:
do we need these prints in production? why?

Reply:
prints were removed